service/controller/resource/azuremachineconditions/conditionvmssready.go

```diff
@@ -0,0 +1,153 @@
+package azuremachineconditions
+
+import (
+	"context"
+
+	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2019-07-01/compute"
+	azureconditions "github.com/giantswarm/apiextensions/v3/pkg/conditions/azure"
+	"github.com/giantswarm/microerror"
+	capz "sigs.k8s.io/cluster-api-provider-azure/api/v1alpha3"
+	capi "sigs.k8s.io/cluster-api/api/v1alpha3"
+	capiconditions "sigs.k8s.io/cluster-api/util/conditions"
+
+	"github.com/giantswarm/azure-operator/v5/service/controller/key"
+)
+
+const (
+	VMSSNotFoundReason                 = "VMSSNotFound"
+	VMSSProvisioningStatePrefix        = "VMSSProvisioningState"
+	VMSSProvisioningStateUnknownReason = "VMSSProvisioningStateUnknown"
+	VmssProvisioningStateSucceeded     = string(compute.ProvisioningStateSucceeded)
```

Comment:
Why different naming scheme? I'd prefer these aligned. No strong opinion which way, your call :slightly_smiling_face:

Reply:
How do you mean different? `VmssProvisioningStateSucceeded` instead of `ProvisioningStateSucceeded`? Can't remember for sure, but I believe I added VMSS prefix, since the reason gets propagated when summarizing or aggregating conditions, so it's easier to see what's happening when this gets propagated to let's say MachinePool, or finally Cluster CR.

P.S. It was also just copypasta from AzureMachinePool :grimacing:  This will soon-ish probably be moved to a separate package where we can do it properly, so I did not put much effort TBH :) Goal here was to do it properly for Cluster API provider-independent conditions, and this one was needed so that `ControlPlaneReady` works by mirroring AzureMachine `Ready`.

Reply:
There are `VMSSProvisioningState.*` constants and then below `VmssProvisioningState.*` constants. My comment here refers to align to either `VMSS` or `Vmss`prefix to all of these constants :slightly_smiling_face:

Reply:
Oh missed that :)